### PR TITLE
Spring boot order

### PR DESCRIPTION
**Pull request by Kulibchenko**

### CODE REVIEW

File: src/main/java/com/example/springboot/model/Order.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    @Column(nullable = false)
+    private BigDecimal total;
+
+    @Column(nullable = false)
+    private LocalDateTime orderDate;
+
+    @Column(nullable = false)
+    private String shippingAddress;
+
+    @OneToMany(mappedBy = "order", fetch = FetchType.EAGER)
```
Comment (Rommelua) [inline src/main/java/com/example/springboot/model/Order.java]: do not use EAGER - why do we need to fetch all items while getting list of Orders?

File: src/main/java/com/example/springboot/service/OrderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        }
+        Set<OrderItem> orderItems;
+        orderItems = cartItems.stream()
+                        .map(item -> {
+                            OrderItem orderItem = new OrderItem();
+                            orderItem.setOrder(order);
+                            orderItem.setBook(item.getBook());
+                            orderItem.setPrice(item.getBook().getPrice());
+                            orderItem.setQuantity(item.getQuantity());
+                            return orderItem;
```
Comment (Rommelua) [inline src/main/java/com/example/springboot/service/OrderServiceImpl.java]: move this to private method

File: src/main/java/com/example/springboot/model/Order.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    public Order() {
+    }
+
+    public enum Status {
+        PENDING,
+        COMPLETED,
+        DELIVERED
+    }
+
```
Comment (koval-kn) [inline src/main/java/com/example/springboot/model/Order.java:62]:
```suggestion
```

File: src/main/java/com/example/springboot/model/Order.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+@SQLDelete(sql = "UPDATE orders SET is_deleted = true WHERE id =?")
+@Where(clause = "is_deleted = false")
+@Table(name = "orders")
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne
+    @JoinColumn(name = "user_id",nullable = false)
```
Comment (koval-kn) [inline src/main/java/com/example/springboot/model/Order.java:35]:
```suggestion
    @JoinColumn(name = "user_id", nullable = false)
```

**Review by koval-kn — APPROVED** [review]

**Pull request merged**